docs/src/main/tut/kleisli.md

```diff
@@ -0,0 +1,237 @@
+
+`Kleisli` can be viewed as the monad transformer for functions. Recall that at its essence, `Kleisli[F, A, B]`
+is just a function `A => F[B]`, with niceties to make working with the value we actually care about, the `B`, easy.
+`Kleisli` allows us to take the effects of functions and have them play nice with the effects of any other `F[_]`.
+
+This may raise the question, what exactly is the "effect" of a function?
+
+Well, if we take a look at any function, we can see it takes some input and produces some output with it, without
+having touched the input (assuming a functional environment). That is, we take a read-only value, and produce some
+value with it. For this reason, the type class instances for functions often refer to the function as a `Reader`.
+For instance, it is common to hear about the `Reader` monad. In the same spirit, Cats defines a `Reader` type alias
+along the lines of:
+
+```tut
+// We want A => B, but Kleisli provides A => F[B]. To make the types/shapes match,
+// we need an F[_] such that providing it a type A is equivalent to A
+// This can be thought of as the type-level equivalent of the identity function
+type Id[A] = A
+
+type Reader[A, B] = Kleisli[Id, A, B]
+object Reader {
+  // Lifts a plain function A => B into a Kleisli, giving us access
+  // to all the useful methods and type class instances
+  def apply[A, B](f: A => B): Reader[A, B] = Kleisli[Id, A, B](f)
+}
+
+type ReaderT[F[_], A, B] = Kleisli[F, A, B]
+val ReaderT = Kleisli
+```
+
+The `ReaderT` value alias exists to allow users to use the `Kleisli` companion object as if it were `ReaderT`, if
+they were so inclined.
+
+The topic of functions as a read-only environment brings us to our next common use case of `Kleisli` - configuration.
+
+### Configuration
+In functional programming it is often advocated to build many small modules, and then compose them together to
+make larger and larger ones until you have the module you want (the overall program). This philosophy (not
+accidentally) mirrors that of function composition - write many small functions, and compose them to build larger ones.
+After all, our programs are just functions.
+
+Let's look at some example modules, where each module has it's own configuration that is validated by a function.
+If the configuration is good, we return a `Some` of the module, otherwise a `None` (you can/should use `Xor` or similar
+to provide more detailed errors - `Option` was chosen for simplicity). Due to the nature of this, we use
+`Kleisli` to define the function.
+
+```tut
+case class DbConfig(url: String, user: String, pass: String)
+trait Db
+object Db {
+  val fromDbConfig: Kleisli[Option, DbConfig, Db] = ???
+}
+
+case class ServiceConfig(addr: String, port: Int)
+trait Service
+object Service {
+  val fromServiceConfig: Kleisli[Option, ServiceConfig, Service] = ???
+}
+```
+
+We have two independent modules, a `Db` and a `Service` (presumably this is a service that will read from a
```

Comment:
I think the example reads better if we just state what things are:

"...a `Db` (allowing access to a database) and a `Service` (supporting an API to provide data over the web)."